### PR TITLE
Allow empty item data set (no orderlines)

### PR DESCRIPTION
**Pull request by Feijs**
It is currently not possible to create or update a transaction without items. This is an issue when trying to remove all items from a cart, and keeping the transaction in sync.

### CODE REVIEW

File: src/Message/ItemDataTrait.php [hunk 1]
```diff
@@ -9,15 +9,15 @@
 trait ItemDataTrait
 {
     /**
-     * @param ItemBag $items
+     * @param ItemBag|null $items
      *
      * @return array[]
      */
-    public function getItemData(ItemBag $items): array
+    public function getItemData(ItemBag $items = null): array
```
Comment (digibeuk) [inline src/Message/ItemDataTrait.php]: I would rather us an empty array as default and avoid the null sitation altogether.

What is the usecase, when exactly is null provided as param?

Comment (Feijs) [reply src/Message/ItemDataTrait.php]: > What is the usecase, when exactly is null provided as param?

> This is an issue when trying to remove all items from a cart, and keeping the transaction in sync.

Comment (Feijs) [reply src/Message/ItemDataTrait.php]: I can change the `AbstractOrderTrait` instead with `'order_lines' => $this->getItemData($this->getItems() ?? new ItemBag()),`

Comment (digibeuk) [reply src/Message/ItemDataTrait.php]: best solution ever! :+1:

**Review by digibeuk — APPROVED** [review]

**Pull request merged**